SConstruct

```diff
@@ -112,6 +112,25 @@ def check_openssl():
             raise Exception("Didn't find any '%s' line in '$ %s'" %
                             (CHECK_LINE, CHECK_COMMAND))
 
+def setImplicitCache():
+    '''Use implicit_cache on some targets to improve build times
+
+    The command line option `--implicit-deps-changed` overrides implicit_cache
+
+    Failure cases are:
+    1) If the include search paths are changed (i.e. CPPPATH), then a file
+       may not be rebuilt.
+    2) If a same-named file has been added to a directory that is earlier in the
+       search path than the directory in which the file was found.
+    Turn on if this build is for a specific debug target (i.e. clang.debug)
+    '''
+    if len(COMMAND_LINE_TARGETS) != 1:
+        return
+    s = COMMAND_LINE_TARGETS[0].split('.')
+    if (len(s) <= 1):
+        return
+    if 'debug' in s:
+        SetOption('implicit_cache', 1)
 
```

Comment:
You don't need parens on 131, but also, these early returns are unneeded in such a short function:

    if len(COMMAND_LINE_TARGETS) == 1:
        s = COMMAND_LINE_TARGETS[0].split('.')
        if len(s) > 1 and 'debug' in s:
            SetOption('implicit_cache', 1)  # Could 1 be replaced by True?